internal/resolution/entitysources/catalogdsource.go

```diff
@@ -0,0 +1,142 @@
+
+	catalogd "github.com/operator-framework/catalogd/pkg/apis/core/v1beta1"
+)
+
+// catalogdEntitySource is a source for(/collection of) deppy defined input.Entity, built from content
+// made accessible on-cluster by https://github.com/operator-framework/catalogd.
+// It is an implementation of deppy defined input.EntitySource
+type catalogdEntitySource map[deppy.Identifier]input.Entity
+
+func NewCatalogdEntitySource(ctx context.Context, client client.Client) (catalogdEntitySource, error) {
+	entities := catalogdEntitySource{}
+	bundleMetadatas, packageMetdatas, err := onClusterMetadata(ctx, client)
+	if err != nil {
+		return nil, err
+	}
+	for _, bundle := range bundleMetadatas.Items {
+		props := map[string]string{}
+
+		for _, prop := range bundle.Spec.Properties {
+			switch prop.Type {
+			case property.TypePackage:
+				type pkg struct {
+					Name    string `json:"packageName"`
+					Version string `json:"version"`
+				}
+				tmp := pkg{}
+				if err := json.Unmarshal(prop.Value, &tmp); err != nil {
+					return nil, err
+				}
+				pkgValue, err := jsonMarshal(tmp)
+				if err != nil {
+					return nil, err
+				}
+				props[property.TypePackage] = string(pkgValue)
+			}
+		}
+
+		imgValue, err := jsonMarshal(bundle.Spec.Image)
+		if err != nil {
+			return nil, err
+		}
+		props["olm.bundle.path"] = string(imgValue)
+		bundlePkg := packageMetdatas[bundle.Spec.Package]
+		channelName := ""
+		for _, ch := range bundlePkg.Spec.Channels {
+			for _, b := range ch.Entries {
+				if b.Name == bundle.Name {
+					channelName = ch.Name
+				}
+			}
+		}
+		channelValue, _ := jsonMarshal(property.Channel{ChannelName: channelName, Priority: 0})
+		props[property.TypeChannel] = string(channelValue)
+		entity := input.Entity{
+			ID:         deppy.IdentifierFromString(fmt.Sprintf("%s%s%s", bundle.Name, bundle.Spec.Package, channelName)),
+			Properties: props,
+		}
+		entities[entity.ID] = entity
+	}
+	return entities, nil
+}
+
+func (es catalogdEntitySource) Get(ctx context.Context, id deppy.Identifier) *input.Entity {
+	if entity, ok := es[id]; ok {
+		return &entity
+	}
+	return nil
+}
+
+func (es catalogdEntitySource) Filter(ctx context.Context, filter input.Predicate) (input.EntityList, error) {
+	resultSet := input.EntityList{}
+	for _, entity := range es {
+		if filter(&entity) {
+			resultSet = append(resultSet, entity)
+		}
+	}
+	return resultSet, nil
+}
+
+func (es catalogdEntitySource) GroupBy(ctx context.Context, fn input.GroupByFunction) (input.EntityListMap, error) {
+	resultSet := input.EntityListMap{}
+	for _, entity := range es {
+		keys := fn(&entity)
+		for _, key := range keys {
+			resultSet[key] = append(resultSet[key], entity)
+		}
+	}
+	return resultSet, nil
+}
+
+func (es catalogdEntitySource) Iterate(ctx context.Context, fn input.IteratorFunction) error {
+	for _, entity := range es {
+		if err := fn(&entity); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func jsonMarshal(p interface{}) ([]byte, error) {
```

Comment:
json.Marshal always escapes for HTML.

Reply:
Sorry! Read right over the fact that you asked specifically about the HTML aspect. HTML encoding subs out `<`, `>`, and `&` for unicode escape sequences, which isn't super human-friendly. In the case of our FBC code in operator-registry (where I suspect @anik120 found this?), this is important because we expect humans to be looking at the output.

In this case, I'm not sure whether humans will ever see anything downstream of this marshaling. Theoretically, I suppose it could show up in resolution error messages?

Reply:
Wouldn't we expect to HTML-escape just before putting the output into a webpage or something? I am still lost as to why we're not operating over raw values here.

Reply:
This is _disabling_ HTML escaping, which is the hardcoded only option in `json.Marshal`. 

However, I think we could probably avoid doing any json marshaling in this entity source. I mentioned in a different comment that there _is_ some impedance mismatch between FBC properties (`[]Property`) and deppy properties (`map[string]string`, I think?), which does seem awkward, but since we already have `json.RawMessage`, it seems like we could do the conversion without needing to marshal/unmarshal anything.

Reply:
We do something like this to disable HTML encoding in `opm render` because we expect humans to be looking at the output.

Here, the resolver will see this. The only place I could see a human being exposed to resolver input is in resolver error messages, and it's conceivable that property values will end up there.

Reply:
I think I'm still really lost. Even in the resolver if I'm reading this code correctly a `json.Unmarshal` happens first so it would be back to raw values ... In user-facing output of JSON (which in and of itself is a bit of an oxymoron) this makes sense but I think unless there's some clear need, starting with a simple call to `json.Marshal` makes sense ...

Reply:
:+1: Seems reasonable to me to start with `json.Marshal` and re-evaluate if there's a need.